PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/arm/ExportTemplate.java

```diff
@@ -47,43 +45,33 @@ public ExportTemplate(DeploymentNode deploymentNode) {
-                String.format(TEMPLATE_FILE_NAME, deploymentNode.getName()));
-        if (file != null) {
-            deploymentNode.getDeployment().exportTemplateAsync().subscribeOn(Schedulers.io()).subscribe(
-                    res -> DefaultLoader.getIdeHelper()
-                            .invokeLater(() -> deploymentNode.getDeploymentNodePresenter()
-                                    .onGetExportTemplateRes(Utils.getPrettyJson(res.templateAsJson()), file)),
-                    ex -> DefaultLoader.getUIHelper().showError(ex.getMessage(), EXPORT_TEMPLATE_FAIL));
-        }
+        String template = deploymentNode.getDeployment().exportTemplate().templateAsJson();
+        String parameters = DeploymentUtils.serializeParameters(deploymentNode.getDeployment());
+        doExport(template, parameters);
     }
 
-    public void doExportParameters() {
-        File file = DefaultLoader.getUIHelper().showFileSaver(PARAMETERS_SELECTOR_TITLE,
-                String.format(PARAMETERS_FILE_NAME, deploymentNode.getName()));
-        if (file != null) {
-            final String parameters = DeploymentUtils.serializeParameters(deploymentNode.getDeployment());
-            deploymentNode.getDeployment().exportTemplateAsync().subscribeOn(Schedulers.io()).subscribe(
-                    res -> DefaultLoader.getIdeHelper()
-                            .invokeLater(() -> deploymentNode.getDeploymentNodePresenter()
-                                    .onGetExportTemplateRes(parameters, file)),
-                    ex -> DefaultLoader.getUIHelper().showError(ex.getMessage(), EXPORT_TEMPLATE_FAIL));
-        }
+    public void doExport(String template, String parameters) {
+        UIUtils.showSingleFolderChooser(FOLDER_SELECTOR_TITLE, (File folder) -> {
+            String templateFile = String.format(TEMPLATE_FILE_NAME, deploymentNode.getName());
+            String parameterFile = String.format(PARAMETERS_FILE_NAME, deploymentNode.getName());
+            deploymentNode.getDeploymentNodePresenter().onGetExportTemplateRes(Utils.getPrettyJson(template),
+                    new File(folder, templateFile), Utils.getPrettyJson(parameters), new File(folder, parameterFile));
+        });
     }
 
-    public void doExport(String template) {
+    public void doExportTemplate(String template) {
         File file = DefaultLoader.getUIHelper().showFileSaver(FILE_SELECTOR_TITLE,
                 String.format(TEMPLATE_FILE_NAME, deploymentNode.getName()));
         if (file != null) {
-            deploymentNode.getDeploymentNodePresenter().onGetExportTemplateRes(Utils.getPrettyJson(template), file);
+            deploymentNode.getDeploymentNodePresenter().onGetExportTemplateFile(Utils.getPrettyJson(template), file);
         }
     }
 
     public void doExportParameters(String parameters) {
         File file = DefaultLoader.getUIHelper().showFileSaver(PARAMETERS_SELECTOR_TITLE,
```

Comment:
Duplicate code of 
```

File file = DefaultLoader.getUIHelper().showFileSaver 
```

Reply:
We design the UIHelper interfaces and use `DefaultLoader.getUIHelper()` to get the implement of IntelliJ, do we need to extract it to a method?